### ISSUE TITLE
can't build lts-10.3 - no such object found when `docker: enable: true`

### ISSUE THREAD

**Issue by silky** [commercialhaskell/stack#4065]
weird error with the latest stack (obtained with `stack upgrade --git`)

```
08:33 AM noon ∈ plot (master) ♫ stack build
Error: No such object: fpco/stack-build:lts-10.3
Received ExitFailure 1 when running
Raw command: /usr/bin/docker inspect fpco/stack-build:lts-10.3
Standard output:

[]

08:33 AM noon ∈ plot (master) ♫ stack --version
Version 1.8.0, Git revision 8ba673f3657c9230e5ada6cee198a7a4842e3775 x86_64 hpack-0.28.2
```

this is the project i'm trying to build - https://github.com/weeezes/plot.git

in particular, if i set `enable: false` [here](https://github.com/weeezes/plot/blob/master/stack.yaml#L53) it works.

**Comment by mgsloan** [comment]
Curious!   No idea why that would not be automatically pulled.  Perhaps try `docker pull fpco/stack-build:lts-10.3`?

**Comment by CodyReichert** [comment]
@mgsloan - fwiw, I got the same error on a newer lts and `docker pull fpco/stack-build:lts-11.22` did work, building with `--docker` was successful afterwards.

**Comment by marnen** [comment]
Having same issue with LTS 12.16 and Docker 18.06.1-ce-mac73.

**Comment by snoyberg** [comment]
Addressed by #4604
